### DISCUSSION TITLE
RELAX NG

### DISCUSSION THREAD

**Discussion by Devasta** [ballsteve/xrust#77] [General]
Hi Steve

(I've not given up on the schematron branch!)

Made some very small steps towards getting RELAX NG added. I found this article

https://relaxng.org/jclark/derivative.html

And I reckon we could use its validation method for both it and DTDs if we get something to build the patterns.

Its going to be a two step process, first to construct the pattern and second to compute the derivative, both are in rust at the moment but there is so much prep work in the first stage I reckon I'll switch it to XSLT.

Far too early to consider merging, but I'll track progress here.

**Comment by ballsteve** [comment]
Hi Daniel,

I’m a bug fan of Relax NG. I used it extensively for a client project a few years ago.

I’ll be interested to see how this goes…

Cheers,
Steve Ball

**Comment by Devasta** [comment]
Just realized I've made my last 2-3 commits directly to the dev branch! I have it in a separate branch (dev_relaxng) so you can overwrite anything I pushed to dev, sorry about that!

**Comment by ballsteve** [comment]
Hi Daniel,

Unless the changes break the build, I wouldn’t worry about it too much.

Let’s treat the current dev as a new baseline, and move forward from there.

Cheers,
Steve
